CMake build fails using XCode generator, blocks universal builds

Trying to build qtkeychain as part of a universal build for macOS - this requires the XCode generator, but using that instead of the usual ninja fails:
```
  CMake Error in 3rdparty/qtkeychain/CMakeLists.txt:
    The custom command generating
  
      /Users/runner/work/Mudlet/Mudlet/3rdparty/qtkeychain/translations/qtkeychain_de.ts
  
    is attached to multiple targets:
  
      qt5keychain
      translations
      messages
  
    but none of these is a common dependency of the other(s).  This is not
    allowed by the Xcode "new build system".
  
  
  CMake Error in test/CMakeLists.txt:
    The custom command generating
  
      /Users/runner/work/Mudlet/Mudlet/3rdparty/qtkeychain/translations/qtkeychain_de.ts
  
    is attached to multiple targets:
  
      qt5keychain
      translations
      messages
  
    but none of these is a common dependency of the other(s).  This is not
    allowed by the Xcode "new build system".
  
  
  CMake Error in 3rdparty/qtkeychain/CMakeLists.txt:
    The custom command generating
  
      /Users/runner/work/Mudlet/Mudlet/3rdparty/qtkeychain/translations/qtkeychain_de.ts
  
    is attached to multiple targets:
  
      qt5keychain
      translations
      messages
  
    but none of these is a common dependency of the other(s).  This is not
    allowed by the Xcode "new build system".
  
  
  CMake Generate step failed.  Build files cannot be regenerated correctly.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -203,7 +203,7 @@ if ( BUILD_TRANSLATIONS )
     qt_create_translation(qtkeychain_MESSAGES ${qtkeychain_TR_SOURCES} ${qtkeychain_TR_FILES})
     qt_add_translation(qtkeychain_QM_FILES ${qtkeychain_TR_FILES})
     add_custom_target(messages DEPENDS ${qtkeychain_MESSAGES})
-    add_custom_target(translations DEPENDS ${qtkeychain_QM_FILES})
+    add_custom_target(translations DEPENDS ${qtkeychain_QM_FILES} messages)
 
     if(QTKEYCHAIN_VERSION_INFIX EQUAL 5 AND QT_TRANSLATIONS_DIR AND NOT QTKEYCHAIN_TRANSLATIONS_DIR)
         # Back compatibility with pre-0.11 versions
@@ -226,6 +226,8 @@ if(NOT QTKEYCHAIN_STATIC)
 else()
     add_library(${QTKEYCHAIN_TARGET_NAME} STATIC ${qtkeychain_SOURCES} ${qtkeychain_MOC_OUTFILES} ${qtkeychain_QM_FILES})
 endif()
+# https://github.com/frankosterfeld/qtkeychain/issues/185
+add_dependencies(${QTKEYCHAIN_TARGET_NAME} translations)
 if(WIN32)
     set_target_properties( ${QTKEYCHAIN_TARGET_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 endif()

